### ISSUE TITLE
0.10.0 regression: false TS1434 on valid {#snippet} / markup (OK on 0.9.20)

### ISSUE THREAD

**Issue by janosh** [pheuter/svelte-check-rs#157]
**Repro**

```bash
git clone https://github.com/janosh/matterviz.git && cd matterviz
npm ci && npx svelte-kit sync
npx svelte-check-rs@0.9.20 --threshold error  # AtomLegend + Structure: no errors
npx svelte-check-rs@0.10.0 --threshold error  # 11 parse errors in those 2 files
```

**0.10.0 output (excerpt)**

```
AtomLegend.svelte:283:60  TS1434  Unexpected keyword or identifier  (×5)
Structure.svelte:1652:23  TS1434  Unexpected keyword or identifier  (×6)
```

**Case A** — blank line before second snippet is reported as TS parse errors on `site_radius_control_snippet`:

```svelte
{/snippet}

{#snippet site_radius_control_snippet()}
```

(`src/lib/structure/AtomLegend.svelte`; needs the full ~236-line `<script>` above it — trimmed snippet-only repro does not fail.)

**Case B** — `{#if x.length > 0}` inside `<button>` after `<Icon />`:

```svelte
<button aria-label="Undo (Cmd/Ctrl+Z)">
  <Icon icon="Undo" />
  {#if undo_stack.length > 0}
    <span>{undo_stack.length}</span>
  {/if}
</button>
```

(`src/lib/structure/Structure.svelte` ~1652; needs ~1600 lines of preceding markup — isolated copy does not fail.)

**Expected:** same as `0.9.20` — no parse errors on valid Svelte 5 source.

**Workaround:** pin `svelte-check-rs@0.9.20`.

**Comment by pheuter** [comment]
Thanks for the detailed repro — reproduced locally and root-caused. Both Case A and Case B turn out to be a **single root cause** in the script's `$props()` destructuring, not the markup.

## Root cause

1. **Latent bug (pre-existing, harmless in 0.9.20):** the `$props()` destructuring parser splits the destructuring body on commas at depth 0 and on the first `:` to handle renames (`prop: local`) — but it is **comment-unaware**. A line comment preceding a prop gets glued onto that prop's chunk, and commas/colons *inside the comment text* are treated as syntax:
   - `AtomLegend.svelte:38` — `// Element remapping: maps original element symbols to new ones` → the colon is misread as a rename, so the "local name" for `element_mapping` becomes `maps original element symbols to new ones\n    element_mapping`.
   - `Structure.svelte:165` — `// ...mapped to H, He, Li by default` → the commas split the comment into bogus "props", leaving `Li by default\n    element_mapping` as a local name. Same for line 167's `// Cell type: original, conventional, or primitive...`.

2. **Trigger (new in 0.10.0, #156):** to mirror upstream svelte2tsx [#3017](https://github.com/sveltejs/language-tools/pull/3017) (suppress TS6133 for unread bindable props under `noUnusedLocals`), the transformer now emits a `;{local_name};` read-marker for every `$bindable()` prop. The comment-polluted local names land verbatim in the generated TypeScript:

   ```ts
   ;atom_color_config;labels;hidden_elements;hidden_prop_vals;maps original element symbols to new ones
       element_mapping;
   ```

   `maps original element symbols ...` is bare invalid tokens → the TS1434/TS1435/TS1005/TS1128 parse errors, mapped back to nearby source positions (which is why Case B *looked* like a markup issue — the positions just map near those template lines).

This also explains why trimmed repros didn't fail: the bug needs the props destructuring **comments**, which the snippet-only repros dropped. Comments *without* a comma/colon are accidentally harmless — the glued `//` prefix keeps the marker line a valid comment.

**Exposure:** any component with a `$bindable()` prop preceded by a line comment containing a `,` or `:` inside the `$props()` destructuring.

## Fix

Make the destructuring scanner comment-aware (skip `//…` and `/*…*/` outside strings) so property chunks never contain comment text. PR incoming.

**Comment by pheuter** [comment]
Fix is up in #158. While validating against matterviz we also found and fixed a second latent issue the parse errors were masking: a workspace path that traverses a symlink broke relative-import resolution in the generated tsgo config (false TS2307s). Details in the PR. With both fixes, the matterviz check matches the 0.9.20 baseline exactly.

**Comment by pheuter** [comment]
Shipped in [v0.10.1](https://github.com/pheuter/svelte-check-rs/releases/tag/v0.10.1) — `npx svelte-check-rs@0.10.1` (or just `@latest`) should now match the 0.9.20 behavior on matterviz. Thanks again for the detailed report!
